app/Http/Controllers/Three/ReactionController.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Rogue\Http\Controllers\Three;
+
+use Rogue\Models\Post;
+use Rogue\Models\Reaction;
+use Illuminate\Http\Request;
+use Rogue\Http\Controllers\Api\ApiController;
+use Rogue\Http\Transformers\Three\ReactionTransformer;
+
+class ReactionController extends ApiController
+{
+    /**
+     * @var \League\Fractal\TransformerAbstract;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->transformer = new ReactionTransformer;
+
+        $this->middleware('auth.api');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Rogue\Models\Post $post
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request, Post $post)
+    {
+        $this->validate($request, [
+            'northstar_id' => 'required|string',
+        ]);
+
+        $userId = $request['northstar_id'];
+        // Check to see if the post has a reaction from this particular user with id of northstar_id.
+        $reaction = Reaction::withTrashed()->where(['northstar_id' => $userId, 'post_id' => $post->id])->first();
+
+        // If a post does not have a reaction from this user, create a reaction.
+        if (is_null($reaction)) {
+            $reaction = Reaction::create([
+                'northstar_id' => $userId,
+                'post_id' => $post->id,
+            ]);
+
+            $code = 200;
+            $action = 'liked';
+        } else {
+            // If the post was previously "liked" by this user, soft delete the reaction. Otherwise, restore the reaction.
+            $code = 201;
+            if (is_null($reaction->deleted_at)) {
+                $action = 'unliked';
+                $reaction->delete();
+            } else {
+                $action = 'liked';
+                $reaction->restore();
+            }
+        }
+
+        $meta = [
+            'action' => $action,
+            'total_reactions' => $this->getTotalReactions($post->id),
+        ];
+
+        return $this->item($reaction, $code, $meta);
+    }
+
+    /**
+     * Returns all reactions for a post.
+     * GET /post/:post_id/reactions
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Rogue\Models\Post $post
+     * @return Illuminate\Http\Response
+     */
+    public function index(Request $request, Post $post)
+    {
+        $query = $this->newQuery(Reaction::class);
```

Comment:
This will return an index of _all_ reactions, which I don't think is what we want for this endpoint. Instead, let's send along the reaction query you'd made below (minus the `get()` that fetches the whole thing):

```diff
- $query = $this->newQuery(Reaction::class);
+ $query = Reaction::withTrashed()->where(['post_id' => $post->id]);
```

Reply:
ohohoh you're right ! Updated :)